pkg/resource/resid.go

```diff
@@ -51,3 +69,19 @@ func (n ResId) Gvk() schema.GroupVersionKind {
 func (n ResId) Name() string {
 	return n.name
 }
+
+// Prefix returns name prefix.
+func (n ResId) Prefix() string {
+	return n.prefix
+}
+
+// SetPrefix set name prefix.
+func (n *ResId) SetPrefix(s string) *ResId {
+	n.prefix = s
+	return n
+}
+
+// NewCopy make a new copy from current ResId
+func (n ResId) NewCopy() ResId {
+	return ResId{gvk: n.gvk, name: n.name, prefix: n.prefix}
+}
```

Comment:
please omit the `SetPrefix` method, and make this method into `CopyWithNewPrefix(p string)`.  Keep things immutable as long as possible.